### ISSUE TITLE
question about Espruino

### ISSUE THREAD

**Issue by tedbeer** [Samsung/iotjs#141]
Do you know about very similar open source Espruino project existing from 2013 - [espruino.com](http://www.espruino.com/Modules), [github.com/espruino](https://github.com/espruino)?
It has same low requirements, it works already on many [boards](http://www.espruino.com/Other+Boards) and it's open source.
Why don't you want to combine efforts? What in Espruino does not meet your needs?

**Comment by lemmaa** [comment]
@tedbeer It's simple. Because the requirements when we first started the project. 

When @egavrin and I imagined JerryScript project in early 2014, we also surveyed many open source projects Duktape, quadwheel, Espruino and even Tessel. But at that time, our 1st target is JS engine which is smaller then 100KB in binary size, and nothing fits it. So we start our own development, internally. 

In other side, we want to know how small we can make a JS engine. So, our first approach is not a full featured JS engine, just small engine which supports subset of JS. As a result, our first version of JerryScript on late 2014 sizes only 93KB and running subset of JS specification. After that, we confirmed that the possibility of JS on embedded systems and plan to grow up JerryScript to full featured one, and open source.

Above all, it started out of curiosity. By accident! So there is no reason to use or not Espruino.
Now and for ever, we will keep trying disruptive experiment with JerryScript. Join us!
Thanks!

P.S. I like Espruino, too. It's really matured project now. :)

**Comment by niutech** [comment]
Really nothing fitted in 100KB? How about [tiny-js](https://github.com/snoozbuster/tiny-js)? In my opinion, it is better to join efforts than to develop yet another JS engine with own APIs, unless they are compatible.

**Comment by lemmaa** [comment]
@niutech  As far as I know, tiny-js is predecessor of Espruino, and it's not designed for full featured. So it's not under our consideration when we first started this project. Sure, we also realized soon that implementing full featured JavaScript engine with full set of built-in ECMA objects under 100KB is not easily possible, so we changed our first goal to subset version. Anyway, we got a great experience with that and now keep going to implement full featured one. As I mentioned in https://github.com/Samsung/iotjs/issues/141, our purpose is not only developing another JavaScript engine, but also keep trying disruptive experiment. That's all. :)
